src/videojs-contrib-hls.js

```diff
@@ -95,6 +95,25 @@ export default class HlsHandler extends Component {
     this.bandwidth = this.options_.bandwidth || 4194304;
     this.bytesReceived = 0;
 
+    // listen for fullscreenchange events for this player so that we
+    // can adjust our quality selection quickly
+    if (document.fullscreenEnabled || document.webkitFullscreenEnabled) {
+      this.on(document, [
```

Comment:
Yeah. I experimented with a *ton* of ways to avoid the infinite event dispatch loop between the player and tech. A number of them technically did the job but they felt grafted-in or added significant capabilities to video.js that I didn't think were warranted. I'm not a huge fan of touching globals but this seems safe & correct to me.